### DISCUSSION TITLE
Guidance - Using one helmrelease to generate multiple helmreleases

### DISCUSSION THREAD

**Discussion by sssd-dev** [fluxcd/flux2#2626] [Q&A] [answered]
We have a use case to use the same helm chart to generate multiple clusters of a stateful application. Using a unique server.id we plan to have multiple such clusters which are going to be using same helm chart in the background. Currently we are using flux v2 to deploy single cluster and would like to have at least 3 such clusters in future.
We would like to avoid have multiple helm releases and overlays and instead would like to explore to have single helm release being templated to generate multiple clusters. Is that a possibility?

I am seeking a way to avoid creating multiple helmreleases. So if there can be a way to create a helmrelease - `name: helmrelease-{{ var }}` and then substitute `var` to generate multiple `helmreleases`. After generating multiple `helmreleases` (on the fly) and then refer to them while applying via overlays.

**Comment by jz-wilson** [comment]
I'm also looking to do the same thing.

**Comment by jprecuch** [comment, marked as answer]
I think you're looking for postBuild which can be applied to kustomize.toolkit and then referenced in HelmRelease: 
https://fluxcd.io/docs/components/kustomize/kustomization/#variable-substitution
https://fluxcd.io/docs/components/kustomize/api/#kustomize.toolkit.fluxcd.io/v1beta2.PostBuild

**Reply by kingdonb** [reply]
This may also help:

https://fluxcd.io/docs/components/helm/helmreleases/#post-renderers

You will need to have multiple helm releases to do it this way, but with the variable substitution approach that @jprecuch proposed, you can have one HelmRelease in the git repository that expands to several different instantiated versions in the cluster. This is closer to what you asked for, I think, but I like the post-renderer function and wanted to make sure you heard about it too.
